File: blatt1/aufgabe1/Caesar.java
public class Caesar {
	static void caesar_chiffre(int k, String msg) {
		char[] letter = msg.toCharArray();
		
		for (int i = 0; i < letter.length; ++i) {
			char crypt = letter[i];
			
			if (crypt >= 'a' && crypt <= 'z') {
				System.out.print((char) ((((crypt - 'a') + k) % 26) + 'a'));
			} else if (crypt >= 'A' && crypt <= 'Z') {
				System.out.print((char) ((((crypt - 'A') + k) % 26) + 'A'));
			} else {
				System.out.print(crypt);
			}
		}
	}
	
	public static void main(String[] args) {
		int k = Integer.parseInt(args[0]) % 26;
		
		if (k < 0) 
			k += 26;
		
		for (int i = 1; i < args.length; ++i) {
			caesar_chiffre(k, args[i]);
			System.out.println();
		}
	}
}

File: blatt1/aufgabe1/caesar.c
#include <stdio.h>
#include <stdlib.h>

void caesar_chiffre(int k, const char *msg);

// ceasar.exe <offset> <to_encode>[]
int main(int argc, const char *argv[]) {
	if (argc < 2) { exit(-1); }
	
	// Stores the defined offset into k
	int k = strtol(argv[1], NULL, 10) % 26;
	
	for (int i = 2; i < argc; i++) {
		caesar_chiffre(k, argv[i]);
		printf("\n");
    }
	
	return 0;
}

void caesar_chiffre(int k, const char *remaining_text){
	while (*remaining_text != '\0') {
		const char crypt = *remaining_text;
		char encrypted = crypt;

		if (crypt >= 'a' && crypt <= 'z') {
			encrypted = (((crypt - 'a') + k) % 26) + 'a';
		} else if (crypt >= 'A' && crypt <= 'Z') {
			encrypted = (((crypt - 'A') + k) % 26) + 'A';
		}

		printf("%c", encrypted);
		remaining_text++;
    }
}